test/extended/openstack/cpms.go

```diff
@@ -81,45 +91,87 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] ControlPlane Mach
 
 		if fds.String() != "" {
 			for _, fd := range objects(controlPlaneMachineSet.Get("spec.template.machines_v1beta1_machine_openshift_io.failureDomains.openstack")) {
-				nova_az := fd.Get("availabilityZone").String()
-				if nova_az != "" {
-					cpmsNovaAvailabilityZones = append(cpmsNovaAvailabilityZones, nova_az)
+				fdMap := make(map[string]string)
+
+				novaAz := fd.Get("availabilityZone").String()
+				cinderAz := fd.Get("rootVolume.availabilityZone").String()
+				volumeType := fd.Get("rootVolume.volumeType").String()
+
+				if novaAz != "" {
+					cpmsNovaAvailabilityZones = append(cpmsNovaAvailabilityZones, novaAz)
+					fdMap["nova_az"] = novaAz
+				}
+				if cinderAz != "" {
+					fdMap["cinder_az"] = cinderAz
+				}
+				if volumeType != "" {
+					fdMap["volume_type"] = volumeType
 				}
+				cpmsFds = append(cpmsFds, fdMap)
+			}
```

Comment:
Consider defining a local FailureDomain type above:

```Go
		type FailureDomain struct {
			ComputeZone string
			VolumeZone  string
			VolumeType  string
		}

		var cpmsFDs []FailureDomain
		var machinesFDs []FailureDomain
```

Then you may be able to compress this whole block into:

```suggestion
				cpmsFDs = append(cpmsFDs, FailureDomain{
					ComputeZone: fd.Get("availabilityZone").String(),
					VolumeZone:  fd.Get("rootVolume.availabilityZone").String(),
					VolumeType:  fd.Get("rootVolume.volumeType").String(),
				})
```